pkg/kubelet/network/dns/dns.go

```diff
@@ -281,28 +283,6 @@ func parseResolvConf(reader io.Reader) (nameservers []string, searches []string,
 	return nameservers, searches, options, utilerrors.NewAggregate(allErrors)
 }
 
-func (c *Configurer) getHostDNSConfig() (*runtimeapi.DNSConfig, error) {
-	var hostDNS, hostSearch, hostOptions []string
-	// Get host DNS settings
-	if c.ResolverConfig != "" {
-		f, err := os.Open(c.ResolverConfig)
-		if err != nil {
-			return nil, err
-		}
-		defer f.Close()
-
-		hostDNS, hostSearch, hostOptions, err = parseResolvConf(f)
-		if err != nil {
-			return nil, err
-		}
-	}
-	return &runtimeapi.DNSConfig{
```

Comment:
Since this was returning blank settings prior to this change for windows, what is going to change now that we are returning the actual host settings?

Reply:
We should create a Release Note regarding this.

So, there are 3 Pod DNS Policies that can be set (https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-s-dns-policy): ``Default``, ``ClusterFirst`` (if not specified, it will be ``ClusterFirst``), and ``ClusterFirstWithHostNet``. I've spawned 3 agnhost pods with all 3 settings (for the last one, I've created a HostProcess Container + host network enabled), while having the ``--resolv-conf=""`` kubelet config option set - which is what we typically set - on a Node that didn't have these changes applied to its kubelet. These were the outputs for their configured DNS suffix lists and DNS nameservers:

```
kubectl exec -n old-ns pod/test-pod-default -- bash -c "/agnhost dns-suffix && /agnhost dns-server-list"
.
127.0.0.1

kubectl exec -n old-ns pod/test-pod-cluster-first -- bash -c "/agnhost dns-suffix && /agnhost dns-server-list"
old-ns.svc.cluster.local,svc.cluster.local,cluster.local
10.96.0.10

kubectl exec -n old-ns pod/test-pod-cluster-first-hostprocess -- ipconfig /all

Windows IP Configuration

   Host Name . . . . . . . . . . . . : test-clus-node01
   Primary Dns Suffix  . . . . . . . :
   Node Type . . . . . . . . . . . . : Hybrid
   IP Routing Enabled. . . . . . . . : No
   WINS Proxy Enabled. . . . . . . . : No
   DNS Suffix Search List. . . . . . : corp.contoso.com
                                       na.corp.contoso.com

Ethernet adapter vEthernet (Ethernet 2):

   Connection-specific DNS Suffix  . : removed.net
   Description . . . . . . . . . . . : Hyper-V Virtual Ethernet Adapter
   Physical Address. . . . . . . . . : 00-22-48-4B-3A-89
   DHCP Enabled. . . . . . . . . . . : Yes
   Autoconfiguration Enabled . . . . : Yes
   Link-local IPv6 Address . . . . . : removed(Preferred)
   IPv4 Address. . . . . . . . . . . : 10.1.0.5(Preferred)
   Subnet Mask . . . . . . . . . . . : 255.255.0.0
   Lease Obtained. . . . . . . . . . : Wednesday, June 8, 2022 8:11:47 AM
   Lease Expires . . . . . . . . . . : Friday, August 25, 2158 7:59:52 PM
   Default Gateway . . . . . . . . . : 10.1.0.1
   DHCP Server . . . . . . . . . . . : 168.63.129.16
   DHCPv6 IAID . . . . . . . . . . . : 167780936
   DHCPv6 Client DUID. . . . . . . . : 00-01-00-01-2A-31-3C-DD-00-0D-3A-75-66-84
   DNS Servers . . . . . . . . . . . : 168.63.129.16
   NetBIOS over Tcpip. . . . . . . . : Enabled
```

Simply setting the ``--resolv-config`` config option to a non-empty string (in my case, I've set it to an empty file path) will result in a few changes for the ``Default`` Pod DNS Policy:

```
# Note that the DNS nameserver list is empty.
kubectl exec -n old-ns pod/test-pod-default -- bash -c "/agnhost dns-suffix && /agnhost dns-server-list"
svc.cluster.local

kubectl exec -n old-ns pod/test-pod-cluster-first -- bash -c "/agnhost dns-suffix && /agnhost dns-server-list"
old-ns.svc.cluster.local,svc.cluster.local,cluster.local
10.96.0.10

# No Change.
kubectl exec -n old-ns pod/test-pod-cluster-first-hostprocess -- ipconfig /all

Windows IP Configuration

   Host Name . . . . . . . . . . . . : test-clus-node01
   Primary Dns Suffix  . . . . . . . :
   Node Type . . . . . . . . . . . . : Hybrid
   IP Routing Enabled. . . . . . . . : No
   WINS Proxy Enabled. . . . . . . . : No
   DNS Suffix Search List. . . . . . : corp.contoso.com
                                       na.corp.contoso.com

Ethernet adapter vEthernet (Ethernet 2):

   Connection-specific DNS Suffix  . : removed.net
   Description . . . . . . . . . . . : Hyper-V Virtual Ethernet Adapter
   Physical Address. . . . . . . . . : 00-22-48-4B-3A-89
   DHCP Enabled. . . . . . . . . . . : Yes
   Autoconfiguration Enabled . . . . : Yes
   Link-local IPv6 Address . . . . . : removed(Preferred)
   IPv4 Address. . . . . . . . . . . : 10.1.0.5(Preferred)
   Subnet Mask . . . . . . . . . . . : 255.255.0.0
   Lease Obtained. . . . . . . . . . : Wednesday, June 8, 2022 8:11:47 AM
   Lease Expires . . . . . . . . . . : Friday, August 25, 2158 7:59:52 PM
   Default Gateway . . . . . . . . . : 10.1.0.1
   DHCP Server . . . . . . . . . . . : 168.63.129.16
   DHCPv6 IAID . . . . . . . . . . . : 167780936
   DHCPv6 Client DUID. . . . . . . . : 00-01-00-01-2A-31-3C-DD-00-0D-3A-75-66-84
   DNS Servers . . . . . . . . . . . : 168.63.129.16
   NetBIOS over Tcpip. . . . . . . . : Enabled
```

Now, with these changes, it depends on how the kubelet is configured. For example, we're typically configuring the kubelet services to run with ``--resolv-config=""``. In this case, this code change will do nothing new, it will still behave as before because of this: https://github.com/kubernetes/kubernetes/blob/3992eda8e61725c470fb6141a7fe4e7f9ee31ea5/pkg/kubelet/network/dns/dns.go#L412 . Now, setting the ``--resolv-config`` config option to a non-empty string value will result in several changes:

- ``ClusterFirst``: has the Host's DNS suffixes lists added at the end.
- ``Default``: has the Host's DNS suffixes added at the start and has the Host's DNS nameserver configured.
- ``ClusterFirstWithHostNet``: no change.

```
kubectl exec -n new-ns pod/test-pod-default -- bash -c "/agnhost dns-suffix && /agnhost dns-server-list"
corp.contoso.com,na.corp.contoso.com,new-ns.svc.cluster.local,svc.cluster.local,cluster.local
168.63.129.16

kubectl exec -n new-ns pod/test-pod-cluster-first -- bash -c "/agnhost dns-suffix && /agnhost dns-server-list"
new-ns.svc.cluster.local,svc.cluster.local,cluster.local,corp.contoso.com,na.corp.contoso.com
10.96.0.10

kubectl exec -n new-ns pod/test-pod-cluster-first-hostprocess -- ipconfig /all

Windows IP Configuration

   Host Name . . . . . . . . . . . . : test-clus-node02
   Primary Dns Suffix  . . . . . . . :
   Node Type . . . . . . . . . . . . : Hybrid
   IP Routing Enabled. . . . . . . . : No
   WINS Proxy Enabled. . . . . . . . : No
   DNS Suffix Search List. . . . . . : corp.contoso.com
                                       na.corp.contoso.com

Ethernet adapter vEthernet (Ethernet 2):

   Connection-specific DNS Suffix  . : removed.net
   Description . . . . . . . . . . . : Hyper-V Virtual Ethernet Adapter
   Physical Address. . . . . . . . . : removed
   DHCP Enabled. . . . . . . . . . . : Yes
   Autoconfiguration Enabled . . . . : Yes
   Link-local IPv6 Address . . . . . : removed(Preferred)
   IPv4 Address. . . . . . . . . . . : 10.1.0.7(Preferred)
   Subnet Mask . . . . . . . . . . . : 255.255.0.0
   Lease Obtained. . . . . . . . . . : Tuesday, June 28, 2022 12:17:12 PM
   Lease Expires . . . . . . . . . . : Friday, August 25, 2158 7:27:43 PM
   Default Gateway . . . . . . . . . : 10.1.0.1
   DHCP Server . . . . . . . . . . . : 168.63.129.16
   DHCPv6 IAID . . . . . . . . . . . : 167780936
   DHCPv6 Client DUID. . . . . . . . : 00-01-00-01-2A-4C-A9-1E-00-0D-3A-75-66-84
   DNS Servers . . . . . . . . . . . : 168.63.129.16
   NetBIOS over Tcpip. . . . . . . . : Enabled
```

Reply:
@jsturtevant are you satisfied with the answer?

@claudiubelu PR is currently marked as release-note-none. Would you please update the PR description to include the release note as per your comment?

Reply:
> @jsturtevant are you satisfied with the answer?
> 
> @claudiubelu PR is currently marked as release-note-none. Would you please update the PR description to include the release note as per your comment?

Updated the Release Note.

Reply:
@jsturtevant - can you comment on this?

Reply:
Yes, this works for me. As the behavior stays the same and it is opt in for different behavior this is a non-breaking change.  It looks like we now have release note as well. Thanks!